cli/cmd/install.go

```diff
@@ -214,6 +213,14 @@ func render(config installConfig, w io.Writer, options *installOptions) error {
 	// Special case for linkerd-proxy running in the Prometheus pod.
 	injectOptions.proxyOutboundCapacity[config.PrometheusImage] = prometheusProxyOutboundCapacity
 
+	if config.EnableProduction {
+		if config.ProxyResourceRequestCPU == "" && config.ProxyResourceRequestMemory == "" {
+			// Override default values when it's production and no CPU or memory is provided
+			injectOptions.proxyCpuRequest = "10m"
+			injectOptions.proxyMemoryRequest = "20Mi"
+		}
```

Comment:
Note that if you make that change, you'll need to tweak the install test so that it uses the options struct that's modified as part of the call to `validateAndBuildConfig`, instead of the `defaultOptions` struct.

```diff
diff --git a/cli/cmd/install_test.go b/cli/cmd/install_test.go
index 2aad3b01..d439a30a 100644
--- a/cli/cmd/install_test.go
+++ b/cli/cmd/install_test.go
@@ -96,13 +96,14 @@ func TestRender(t *testing.T) {
 
 	testCases := []struct {
 		config                installConfig
+		options               *installOptions
 		controlPlaneNamespace string
 		goldenFileName        string
 	}{
-		{*defaultConfig, defaultControlPlaneNamespace, "testdata/install_default.golden"},
-		{metaConfig, metaConfig.Namespace, "testdata/install_output.golden"},
-		{singleNamespaceConfig, singleNamespaceConfig.Namespace, "testdata/install_single_namespace_output.golden"},
-		{*productionConfig, productionConfig.Namespace, "testdata/install_production_output.golden"},
+		{*defaultConfig, defaultOptions, defaultControlPlaneNamespace, "testdata/install_default.golden"},
+		{metaConfig, defaultOptions, metaConfig.Namespace, "testdata/install_output.golden"},
+		{singleNamespaceConfig, defaultOptions, singleNamespaceConfig.Namespace, "testdata/install_single_namespace_output.golden"},
+		{*productionConfig, productionOptions, productionConfig.Namespace, "testdata/install_production_output.golden"},
 	}
 
 	for i, tc := range testCases {
@@ -110,7 +111,7 @@ func TestRender(t *testing.T) {
 			controlPlaneNamespace = tc.controlPlaneNamespace
 
 			var buf bytes.Buffer
-			err := render(tc.config, &buf, defaultOptions)
+			err := render(tc.config, &buf, tc.options)
 			if err != nil {
 				t.Fatalf("Unexpected error: %v", err)
 			}
```